be/src/http/download_action.cpp

```diff
@@ -254,7 +258,8 @@ Status DownloadAction::check_path_is_allowed(const std::string& file_path) {
 
 Status DownloadAction::check_log_path_is_allowed(const std::string& file_path) {
     DCHECK_EQ(_download_type, ERROR_LOG);
-    if (FileSystemUtil::contain_path(_error_log_root_dir, file_path)) {
+    std::string canonical_file_path = canonical(file_path).string();
+    if (FileSystemUtil::contain_path(canonical(_error_log_root_dir).string(), canonical_file_path)) {
```

Comment:
`_error_log_root_dir` should be set after canonical when initializing.

Reply:
why not just call canonical in DownloadAction construct method?